### PR TITLE
Improve check repositories check

### PR DESCRIPTION
**Pull request by wlach**
tl;dr: uses the repository parser instead of hand reading the YAML, which improves forward compatibility.

### CODE REVIEW

**Comment by wlach** [comment]
I would not normally bother with this kind of cleanup, but the existing checks failed when I tried to do some library refactoring.

**Comment by wlach** [comment]
I verified by hand that this produces the errors it's expected to, using this diff:

```diff
diff --git a/repositories.yaml b/repositories.yaml
index 8abb863..f2bef49 100644
--- a/repositories.yaml
+++ b/repositories.yaml
@@ -14,7 +14,7 @@ libraries:
     url: https://github.com/mozilla/glean
     branch: main
     metrics_files:
-      - glean-core/metrics.yaml
+      - glean-cores/metrics.yaml
     ping_files:
       - glean-core/pings.yaml
     library_names:
@@ -210,7 +210,6 @@ applications:
       - v1_name: firefox-desktop
         app_id: firefox-desktop
 
-
   - app_name: fenix
     app_description: Firefox for Android (Fenix)
     canonical_app_name: Firefox for Android
@@ -230,6 +229,13 @@ applications:
       - org.mozilla.components:support-migration
       - org.mozilla.components:places
     channels:
+      - v1_name: firefox-android-release-2
+        app_id: org.mozilla.firefox
+        app_channel: release
+        additional_dependencies:
+          - org.mozilla.components:browser-engine-gecko-beta
+        description: >-
+          Release channel of Firefox for Android.
       - v1_name: firefox-android-release
         app_id: org.mozilla.firefox
         app_channel: release
```

File: probe_scraper/check_repositories.py [hunk 1]
```diff
@@ -1,69 +1,75 @@
...
-                + branch
-                + "/"
-                + metric_file
+    if repo.app_id and repo.channel and not repo.deprecated:
+        app_id_channels[repo.app_id][repo.channel] += 1
+
+    for metric_file in metrics_files:
+        temp_url = (
+            GITHUB_RAW_URL
+            + repo.url.replace("https://github.com", "")
```
Comment (fbertsch) [inline probe_scraper/check_repositories.py:29]:
```suggestion
            repo.url.replace("https://github.com", GITHUB_RAW_URL)
```

**Review by fbertsch — APPROVED** [review]
r+wc, thanks for cleaning this up!

**Pull request merged**